x-pack/plugin/mapper-aggregate-metric/src/main/java/org/elasticsearch/xpack/aggregatemetric/mapper/AggregateDoubleMetricFieldMapper.java

```diff
@@ -677,24 +672,20 @@ protected void parseCreateField(DocumentParserContext context) throws IOExceptio
             }
         } catch (Exception e) {
             if (ignoreMalformed) {
-                if (malformedContentForSyntheticSource != null) {
-                    if (subParser != null) {
-                        // Remaining data in parser needs to be stored as is in order to provide it in synthetic source.
-                        XContentHelper.drainAndClose(subParser, malformedContentForSyntheticSource);
-                    } else {
-                        // We don't use DrainingXContentParser since we don't want to go beyond current field
-                        malformedContentForSyntheticSource.copyCurrentStructure(context.parser());
-                    }
-                    ;
-                    var nameValue = IgnoredSourceFieldMapper.NameValue.fromContext(
-                        context,
-                        name(),
-                        XContentDataHelper.encodeXContentBuilder(malformedContentForSyntheticSource)
-                    );
-                    context.addIgnoredField(nameValue);
-                } else if (subParser != null) {
+                if (subParser != null) {
                     // close the subParser, so we advance to the end of the object
                     subParser.close();
+                } else {
+                    if (context.mappingLookup().isSourceSynthetic()) {
+                        // There is a malformed value, but it is not an object (since subParser is null).
+                        // So we just need to copy this single value.
+                        malformedDataForSyntheticSource = XContentBuilder.builder(context.parser().contentType().xContent())
+                            .copyCurrentStructure(context.parser());
+                    }
+                }
+
+                if (malformedDataForSyntheticSource != null) {
+                    context.doc().add(IgnoreMalformedStoredValues.storedField(name(), malformedDataForSyntheticSource));
                 }
```

Comment:
This blocks gets repeated, consider moving it to a helper in `IgnoreMalformedStoredValues`

Reply:
I'll leave it if you don't mind i don't think it's too bad of a repetition.